### DISCUSSION TITLE
posts/sqlclient-active-directory-authent

### DISCUSSION THREAD

**Discussion by giscus** [TechWatching/techwatching.dev#38] [Announcements]
# posts/sqlclient-active-directory-authent

https://www.techwatching.dev/posts/sqlclient-active-directory-authent

**Comment by drmcclelland** [comment]
During local development of a .NET 6 API, I have also been trying to use the new "Active Directory Default" authentication method (as you so helpfully detailed above), but I still keep getting this error message:

Microsoft.Data.SqlClient.SqlException (0x80131904): Login failed for user '<token-identified principal="">'. The server is not currently configured to accept this token.

Any idea where I can check for this particular configuration?

The only "passwordless" method that seems to work for me is "Active Directory Interactive"...   but even though it works, that's not truly passwordless, it pops up a browser to authenticate me.

**Reply by drmcclelland** [reply]
Found the culprit:  in my Visual Studio 2022 "profile" menu, it had a prompt saying "Re-enter your credentials".  Once I did that, this method started working just fine!

**Reply by TechWatching** [reply]
Great to see you made it work. When I have some issues like that with account logged in visual studio, sometimes I simply use Azure CLI  to log in.
